socorro/external/boto/connection_context.py

```diff
@@ -202,6 +206,9 @@ def submit(self, id, name_of_thing, thing):
         key_object = bucket.new_key(key)
         key_object.set_contents_from_string(thing)
 
+        elapsed_time = time.time() - start_time
+        self.metrics.histogram(name_of_thing, value=elapsed_time * 1000.0)
```

Comment:
I'd add "submit" to this, too. It'll make it easier to distinguish other activities from one another if we add timings for other things later.

So, something like `'submit.' + name_of_thing` if we want the hierarchy to be processor -> boto -> submit -> raw_crash. Or `name_of_thing + '.submit'` if we want the hierarchy to be processor -> boto -> raw_crash -> submit. I'm on the fence about which is better for our purposes. Maybe the latter.

@sciurus Do you have a preference for the hierarchy? Is one more convenient with your Datadog graphing plans?

Reply:
I'd prefer that `name_of_thing` not be in the hierarchy at all, and instead be recorded via a tag. You run into gray areas pretty quickly, but generally I think a metric's name should reflect the service (socorro processor) and the operation (submitting to s3) and tags should be used for any additional dimensions (e.g. what you're submitting).

The key thing with Datadog is that tags, not the hierarchy encoded in the name,  determine how you can aggregate. If we put the type of thing being submitted in the metric's name, it's harder to analyze the timing of all submissions. If we put it in a tag, it's easy to look at all submissions or to break it down to a specific type.

Reply:
The "name_of_thing" is a finite and small set of things, so that makes sense to me and would work here nicely.

@AlexisDeschamps Can you change this to use a tag? It'd be something like this:

```
self.metrics.histogram('submit', value=elapsed_time * 1000.0, tags=['type:' + name_of_thing])
```

I used "type" there, but "kind", "genus", or any word like that is fine.

Reply:
`type` in particular isn't a good tag key, because the metric will already have a `Type` tag that is picked up from AWS.

https://mana.mozilla.org/wiki/display/SVCOPS/AWS+EC2+Instance+Tags